Add meeting minutes for March 2023 (#39)
ocapn/ocapn: meeting-minutes/2023-03-24.md

# March 2023 OCapN pre-standardization meeting, 2023-03-24

- **Chair:** Jonathan Rees
- **Scribe:** Christine Lemmer Webber
- **Present:**
  - Alan Karp (alan)
  - Baldur (Zarutian)
  - Christine Lemmer Webber (cwebber)
  - David Thompson (dthompson)
  - Dan Connolly (dckc)
  - Ian Denhardt (isd)
  - Jessica Tallon (Jessica)
  - Jonathan Rees (jar286)
  - Kris Kowal (kriskowal)
  - Mark Miller (markm)
  - Juliana Sims (juliana)

> 

Reviewer: - Jacob Weisz (ocdtrekkie)

I was silent, but present.

Jessica: starting with status report on draft specifications

Jessica: writing three documents

Jessica: main document is document specifying CapTP, starting and breaking a connection, promises, promise pipelining, 3rd party handoffs

Jessica: a smaller amount is netlayers, opening a channel through a particular transport protocol, at first it'll be the TOR onion netlayer, but obviously other netlayers will be added and implementations can pick and choose their netlayers. that covers opening a secure channel between two machines

Jessica: third one is the "OCapN URI" spec, maybe controversial now, represents sturdyrefs, how to establish in-bound and out-of-bound locations

Jessica: that would be the three main documents, the fourth may not be included, but is the Syrup spec, which is the serialization form based on preserves

Jessica: mainly written with a draft README, since the specs will *initially* based on Spritely's implementation, not to favor it, but as a starting point, but the group can take these draft specifications that we were writing

Jessica: and then the group can evolve as the group reaches decisions

> Jessica: ETA on those, the Syrup one is ready for review, is ready as a PR on the syrup spec

Reviewer 2: I believe we did note the URL of the PR during the meeting, so...

```suggestion
Jessica: ETA on those, the Syrup one is ready for review, is ready as [a PR on the syrup spec](https://github.com/ocapn/syrup/pull/9)
```

isd: so in terms of is to whether the abstract syntax is specified, I'd look at Preserves, so Syrup is just one encoding for Preserves

isd: one thing I'm fuzzy on is what agreement is here

isd: I'm curious how much Goblins to bend on this

isd: it's clear we can't get capn'proto and agoric to agree, but not sure we can't get spritely to bend to agoric's, I want to get clear as in terms of a scope thing

isd: is it something that's on the table?  or have we already decided that's not feasible and we're trying to see whether

Jessica: I think spritely can make some changes re: captp, but as relative newcomer to the group I don't have Agoric's CapTP spec in my head

dckc: would like to know about interop

cwebber: I think if Jessica can have abstract stuff "as separate as possible", and we *aim* for as close as possible compatibility between agoric and spritely, and not decide whether it's possible to get full compatibility *yet*

jar286: can we get a concrete action item?

jar286: going to create an issue about this topic

# TOPIC: Agoric/Spritely interop status report Demonstrate Agoric & Spritely interop #6; next steps (@dckc)

jar286: next topic is interoperability

> jar286: dckc has the floor
> 

Reviewer 2: let's skip stage direction such as this in the interest of conciseness

```suggestion
```

> dckc: this is a writeup from talking with Kris Kowal

Reviewer 2: ```suggestion
dckc: [this demo sketch](https://github.com/ocapn/ocapn/issues/6#issuecomment-1477952429) is a writeup from talking with Kris Kowal and company
```

dckc: here spritely has a chat, spritely has spritely chat on one side, so spritely goes over a unix domain socket plugin which then the endo plugin for sockets, and then a captp bridge 

dckc: and then a js chat app with an http endo plugin and a UI which goes on the Agoric side

cwebber: I think this is an interesting goal towards trying to show off how this would work

> dckc: small amount of hacking I did do was about the coordinated types

Reviewer 2: ```suggestion
dckc: [small amount of hacking](https://github.com/ocapn/ocapn/issues/5#issuecomment-1478965669) I did do was about the core data types
```

dckc: would like to see early stuff to the group

Jessica: in terms of asking for broader review, it may be in an earlier pre-draft state than I had hoped but I could

isd: I think dckc's comment adds some interesting questions, eg what can we do with low-level collections

isd: those layers are not separated out in syrup/preserves

isd: I think one of the questions is whether or not we can move those two things together

isd: how attached are we from what parts of this data level

isd: could you look at agoric's and see what parts we do and don't want to adopt?

isd: I think having that conversation is the first order of businsess in terms of whether or not we can converge on a shared protocol

markm: dckc said mapping a subset to a subset... if there's a joint subset that round trips, I'd describe that as a subset we've agreed upon.  so the layer separation we have now, that was partly inspired by one of these conversations.  I was trying to put collections in at the arrays/records stuff.  and I had to write down a lot of semantics of what it meant for two keys to be equal.  for example, if they're keys in a map or a set and

> 

Reviewer 2: ```suggestion

https://github.com/ocapn/ocapn/issues/5#issuecomment-1478684631 is a pre-meeting comment from markm to largely the same effect

```

markm: it's still recognized as a tag, but not recognized as having only higher level meaning

cwebber: preserves alone is not expressive enough, just acking, because then we couldn't represent the liverefs over the wire

cwebber: but tagged data structures, that's part of the history of how preserves came about

jar286: who's shepherding this issue?

isd: I can manage, spritely folks can you look at the agoric's stuff

jar286: I did make a wiki page of my thoughts as chair, want to make people aware

jar286: I like assigning people to issues to find out next step

dckc: if next step on data model is to prompt spritely folks, I'd like to prompt spritely folks

cwebber: tsyeika did an incredible job on this doc, encourage people to look at it

dckc: if it's useful to have two people assigned, the interop demo is going to involve going from one thing to another, so I'm willing to be assigned as well

jar286: 3 minutes left, I liked the traidtion we had in the w3c of doing issue review

jar286: a drive to close issues or assign to people, so is there anything else we need to cover

> jar286: looks like Tuesday April 25 at 19:00 UTC (next meeting)

Reviewer 2: ```suggestion
# Schedule for next monthly meeting

jar286: looks like Tuesday April 25 at 19:00 UTC (next meeting)
```

Hm... all but the 1st heading should be h2, i.e. `##`.
